batch/test/basics.js

```diff
@@ -69,6 +61,28 @@ describe('Create, list and delete jobs', () => {
     assert(output !== null);
   });
 
+  it('get task', async () => {
+    // the server needs a bit of time to create the objects. 10 seconds is way more than enough.
+    await new Promise(resolve => setTimeout(resolve, 10000));
```

Comment:
This will just delay the one test, right? In my experience tests sometimes run concurrently (well, async-concurrently) so you might have an issue with 'list tasks' below, too. I'm not super familiar with this code.

Reply:
I've checked the docs and the only concurrency I've found was for running different test files in parallel, in different processes. It was added in mocha v8.0.

Could you point me to an example of this async-concurrency happening? I'd love to use it if it's available, although I'd probably have to merge many of the individual tests we have now into bigger tests to support concurrent execution.

Reply:
It's entirely anecdotally from working on the Node Pub/Sub library, I found that if I didn't use unique topic/subscription names for each test (guid or whatnot), they would sometimes step on each other. Admittedly that might've been an artifact of the service itself lagging on creating and deleting objects, but my understanding of it is that, when one of your tests hits `await`, it'll swap off to something else (might be another test). I'm not sure, it's worth looking into probably... like you say, it'd be useful to have explicit control over that. In general though, if you can, it's good to try to make the tests independent so that you can e.g. run one of them selectively with `-f`.

Reply:
I've been asked to do the exact opposite for Go samples so I'm really quite confused now :laughing:

Reply:
But according to both my tests and mocha documentation, tests are never run async-concurrently.